src/Identity/Core/src/SignInManager.cs

```diff
@@ -432,6 +464,312 @@ public virtual async Task<SignInResult> CheckPasswordSignInAsync(TUser user, str
+    /// for the sign-in attempt.
+    /// </returns>
+    public virtual async Task<SignInResult> PasskeySignInAsync(string credentialJson, PasskeyRequestOptions options)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(credentialJson);
+
+        var assertionResult = await PerformPasskeyAssertionAsync(credentialJson, options);
+        if (!assertionResult.Succeeded)
+        {
+            return SignInResult.Failed;
+        }
+
+        var setPasskeyResult = await UserManager.SetPasskeyAsync(assertionResult.User, assertionResult.Passkey);
+        if (!setPasskeyResult.Succeeded)
+        {
+            return SignInResult.Failed;
+        }
+
+        return await SignInOrTwoFactorAsync(assertionResult.User, isPersistent: false, bypassTwoFactor: true);
+    }
+
+    /// <summary>
+    /// Generates a <see cref="PasskeyCreationOptions"/> and stores it in the current <see cref="HttpContext"/> for later retrieval.
+    /// </summary>
+    /// <param name="creationArgs">Args for configuring the <see cref="PasskeyCreationOptions"/>.</param>
+    /// <returns>
+    /// A task object representing the asynchronous operation containing the <see cref="PasskeyCreationOptions"/>.
+    /// </returns>
+    public virtual async Task<PasskeyCreationOptions> ConfigurePasskeyCreationOptionsAsync(PasskeyCreationArgs creationArgs)
+    {
+        ArgumentNullException.ThrowIfNull(creationArgs);
+
+        var options = await GeneratePasskeyCreationOptionsAsync(creationArgs);
+
+        var props = new AuthenticationProperties();
+        props.Items[PasskeyCreationOptionsKey] = options.AsJson();
+        var claimsIdentity = new ClaimsIdentity(new ClaimsIdentity(IdentityConstants.TwoFactorUserIdScheme));
+        claimsIdentity.AddClaim(new Claim(ClaimTypes.NameIdentifier, options.UserEntity.Id));
+        claimsIdentity.AddClaim(new Claim(ClaimTypes.Email, options.UserEntity.Name));
+        claimsIdentity.AddClaim(new Claim(ClaimTypes.Name, options.UserEntity.DisplayName));
+        var claimsPrincipal = new ClaimsPrincipal(claimsIdentity);
+        await Context.SignInAsync(IdentityConstants.TwoFactorUserIdScheme, claimsPrincipal, props);
+
+        return options;
+    }
+
+    /// <summary>
+    /// Generates a <see cref="PasskeyCreationOptions"/> to create a new passkey for a user.
+    /// </summary>
+    /// <param name="creationArgs">Args for configuring the <see cref="PasskeyCreationOptions"/>.</param>
+    /// <returns>
+    /// A task object representing the asynchronous operation containing the <see cref="PasskeyCreationOptions"/>.
+    /// </returns>
+    public virtual async Task<PasskeyCreationOptions> GeneratePasskeyCreationOptionsAsync(PasskeyCreationArgs creationArgs)
+    {
+        ArgumentNullException.ThrowIfNull(creationArgs);
+
+        var excludeCredentials = await GetExcludeCredentialsAsync();
+        var serverDomain = Options.Passkey.ServerDomain ?? Context.Request.Host.Host;
```

Comment:
In a previous iteration of this PR, the computed domain was configurable via custom `IPasskeyRequestContextProvider` implementation. I've since removed that abstraction because it became less necessary after moving most of the passkey implementation to the `Microsoft.AspNetCore.Identity` package. Curious if others think we need to allow customization of the "fallback" logic here in the case of `Options.Passkey.ServerDomain` not being specified.

Reply:
I think it's ok. And people can already do this by overriding this method, if necessary, can't they? It's a matter of how hard it is, not how possible it is.